### DISCUSSION TITLE
disable floating cmdline

### DISCUSSION THREAD

**Discussion by verajosemanuel** [LazyVim/LazyVim#2312] [Q&A] [answered]
Hi I'm not very fond of the floating cmdline, so I tried to disable it. My first idea was to override config in options adding a line:
`opts.cmdline.view = 'cmdline'
`
but that did not the trick. The next thing I tried is a new file in plugins folder called noice_config.lua with this content (but Lazyvim crashed):
```
return {{
  "folke/noice.nvim",
  event = "VeryLazy",
  opts = {
    lsp = {
      override = {
        ["vim.lsp.util.convert_input_to_markdown_lines"] = true,
        ["vim.lsp.util.stylize_markdown"] = true,
        ["cmp.entry.get_documentation"] = true,
      },
    },
    routes = {
      {
        filter = {
          event = "msg_show",
          any = {
            { find = "%d+L, %d+B" },
            { find = "; after #%d+" },
            { find = "; before #%d+" },
          },
        },
        view = "mini",
      },
    },
    presets = {
      bottom_search = true,
      command_palette = false,
      long_message_to_split = false,
      inc_rename = false,
    },
  },
  -- stylua: ignore
  keys = {
    { "<S-Enter>", function() require("noice").redirect(vim.fn.getcmdline()) end, mode = "c", desc = "Redirect Cmdline" },
    { "<leader>snl", function() require("noice").cmd("last") end, desc = "Noice Last Message" },
    { "<leader>snh", function() require("noice").cmd("history") end, desc = "Noice History" },
    { "<leader>sna", function() require("noice").cmd("all") end, desc = "Noice All" },
    { "<leader>snd", function() require("noice").cmd("dismiss") end, desc = "Dismiss All" },
    { "<c-f>", function() if not require("noice.lsp").scroll(4) then return "<c-f>" end end, silent = true, expr = true, desc = "Scroll forward", mode = {"i", "n", "s"} },
    { "<c-b>", function() if not require("noice.lsp").scroll(-4) then return "<c-b>" end end, silent = true, expr = true, desc = "Scroll backward", mode = {"i", "n", "s"}},
  },
}}
```

which is the proper way of having the classic cmdline at the bottom?

**Comment by dpetka2001** [comment, marked as answer]
#1275 and #1926 should be of help.

**Reply by verajosemanuel** [reply]
Nearly. I set cmdline & popmenu to false but messages to true as my guess is I could have just the cmdline at bottom . Wrong guessing. Lazyvim stuck on a message and unresponsive. So I have to disable all. Maybe a clean way is disabling noice completely.

**Reply by dpetka2001** [reply]
You also have to disable messages as well. From the docs
>-- NOTE: If you enable messages, then the cmdline is enabled automatically.
    -- This is a current Neovim limitation.

**Comment by boydkelly** [comment]
so is there any way to customize noice?   I would like to customize the command line.   I would want to specify the position, border, winhighlight etc, as well as some other presets.

**Reply by dpetka2001** [reply]
You should read the docs of noice then to see if you can do what you want. Never done this myself.

**Comment by boydkelly** [comment]
Ah... i aready have customized it in my current config.   But i see i just had to copy the setup piece into plugins...
return {
    {
        "folke/noice.nvim",
        event = "VimEnter",
        -- event = "VeryLazy",
        config = function()
            local icons = require("lazyvim.config").icons.kinds
            require("noice").setup({
                debug = false,
                presets = {
                    bottom_search = false,
                    command_palette = true,
                    long_message_to_split = true,
                    inc_rename = false,
                    cmdline_output_to_split = false,
                    lsp_doc_border = true,
                },
                cmdline = {
                    enabled = true,
                    view = "cmdline",
                    format = {
